src/fluree/db/db/json_ld.cljc

```diff
@@ -195,34 +186,30 @@
 
 (defn retrieve-context
   "Returns the parsed context. Caches."
-  [default-context context-cache supplied-context context-type]
-  (log/trace "retrieve-context - default: " default-context "supplied:" supplied-context "context-type: " context-type)
-  (or (get-in @context-cache [context-type supplied-context])
+  [default-context context-cache supplied-context]
+  (log/debug "retrieve-context - default: " default-context "supplied:" supplied-context)
```

Comment:
The problem I have with it is that debug is enabled by default in development, so the repl can sometimes become unusable when you have a running system that you are exploring. Also, I view trace logs as literally tracing through the execution of the code, and that's more in line with what's happening here (admittedly subjective).

I would just rather not have all these messages printed to the repl by default. That makes it more difficult to actually identify the things you care about when in development.

Reply:
I think INFO should be the default, I for one am often drowning in the DEBUG stuff but I've been too lazy to adjust. Some sort of normalization of log levels should be done, though, I think that would make logging more useful for everyone.

Reply:
I agree about not cluttering the REPL. But debug being enabled is just our dev logging config, isn't it? We can change that whenever we want. It's probably the `dev/logback.xml` file.

Reply:
I think for me these specific logs are too granular, more trace-like. I would be inclined to change my logging level to omit these, but then I would lose others that I consider more valuable.

Fwiw for me debug logs are more "top-level," like at the entry or exit points of namespaces or significant code paths. I use them to help me figure out where to start looking if something has gone wrong.

These logs are ones I'd want enabled if, for example, I knew I needed to be looking in this ns and following its operations closely. To me that's more trace-level.

Reply:
Yeah that all seems reasonable. I just would like to have a shared, documented definition to follow. Otherwise it's easy to get lost in "what level do I need where?"

For this PR I'm happy to remove any debug logs that seem to create more noise than they're worth.

Reply:
DEBUG is really a customer-driven requirement first. While developing you are probably doing 1 operation a minute, but customers will easily have 1,000,000x more volume.

If there is an issue we cannot immediately resolve, our next step is to ask them to set logging to DEBUG. This needs to be done so they don't immediately run out of disk space, or blow up their logging software costs. Importantly it needs to be useful but also manageable for us to analyze.

TRACE is last resort, and likely a system can only run in TRACE for a very short amount of time.

Importantly all logging should take care not to leak sensitive information (customer data). This is one of the top ways data breaches happen.

Reply:
I said "default log level", which I now get is confusing. I did not mean the default log level as set in logback.xml. I meant the log level we should usually pick when adding a new log statement in our code. 

I think Marcela and Brian said better what I was trying to get at. I view the TRACE level as a granular, function by function, tracing through each step of code execution. That's useful sometimes, but when it's on all the time, it becomes useless. I view DEBUG as high level view when specific components are engaged (like starting to search for a query with a specific where clause), which is almost always useful in development, and also useful to figure out what components are breaking in production for a customer. I vew INFO as essential messages reporting out progress or normal operations of the program that would be useful in production and for a customer to monitor. WARN is for something the customer should watch out for, and ERROR is for when something failed.  

The log levels for the program are set in the logback.xml. We can edit that file in development when we want to change the log level for the whole program or a specific namespace when we want more or less messages. The level is debug in development and info in production (as I think it should be).  

These function by function logs printing out the input and output of each function should be set to trace, in my opinion. Then, for the few cases in development or wherever that we actually want such granular logs when we're trying to pinpoint an issue in a specific function or verify that it's doing the right thing, we can change the logback.xml to set the log level to trace for _just the namespace we care about at that time_. Using the same level for everything removes that flexibility.

Reply:
This all seems fine but we should specify and document it so we can all be on the same page about what to use when in the future.